src/components/CardModal/CardModalActions/AttachmentMenu/AttachmentMenu.js

```diff
@@ -0,0 +1,69 @@
+import localforage from "localforage";
+import React, { useState, useRef } from "react";
+import { MdClose } from "react-icons/md";
+import { useDispatch, useSelector } from "react-redux";
+import { closeMenu } from "../../../../features/modalActionMenusVisibilitySlice";
+import { v4 as uuidv4 } from "uuid";
+import { addFilesToCard } from "../../../../features/cardsSlice";
+import { setCurrentSelectedCard } from "../../../../features/currentSelectedCardSlice";
+
+const AttachmentMenu = ({ update }) => {
+  const hiddenFileInput = useRef(null);
+  const dispatch = useDispatch();
+
+  let style = "fixed mt-10 w-72 text-gray-700 bg-white rounded-ibsm shadow-2xl p-4"
+  if (update) style = "fixed w-72 text-gray-700 bg-white rounded-ibsm shadow-2xl p-4"
+
+  const handleFileSelect = async(e) => {
+      const files = e.target.files;
+      let upload = []
+      if (files) {
+          for (let i= 0; i < files.length; i++) {
+            let id = uuidv4();
+            let date = JSON.stringify(new Date());
+            await localforage.setItem(id, files[i])
+            upload.push({fileId: id, date, name: files[i].name});
+          }
+      }
+      console.log('upload',upload)
+
+      dispatch(
+          addFilesToCard (
+              { 
+                  cardId: card.id, 
+                  upload, 
+              }
+          )
+      );
+      dispatch(setCurrentSelectedCard({ ...card, attachments: [...card.attachments, ...upload] }));
+      handleClose();
+  }
+
+  const card = useSelector((state) => state.currentSelectedCard.value);
```

Comment:
This should be at the top with `hiddenFileInput` and `dispatch`.

Reply:
Components should generally have variables near the top, then methods, then the layout. It's a minor thing but helps readability and consistency

Reply:
Remove console.log please.